ara_enumerables.rb

```diff
@@ -0,0 +1,129 @@
+# my enumerable
+require './utils.rb'
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
```

Comment:
#### [optional]
Pass the method name as an argument to ```to_enum``` like you have done for my_each to make your method more customized.
``` return to_enum(:my_select) unless block_given?```